### DISCUSSION TITLE
GPIO power on button

### DISCUSSION THREAD

**Discussion by alpapak** [PiSupply/PiJuice#713] [Q&A] [answered]
I would like to add a power on button on my Project. 

My set up is :
Pi4 with USB-C power cable
PiJuice hat

I am trying to add the button in GPIO3(pin 5) and GROUND(pin 6).

When my Pi4 has power for USB-C it is working.
When it is power from PiJuice it is not.
Why? Can i fix it?

**Comment by tvoverbeek** [comment, marked as answer]
The boot via GPIO3 is performed by the videocore firmware. Hence when the Pi is shut down the 5V to the Pi still needs to be present to keep the videocore alive. If you power via the PiJuice then make sure the 5V from the PiJuice to the Pi is not removed at shutdown(e.g. use SYS_FUNC_HALT, not SYS_FUNC_HALT_POW_OFF). Of course this causes more battery drain than with the 5V removed.

**Reply by alpapak** [reply]
I was using SYS_FUNC_HALT_POW_OFF. Thank you.
